### PR TITLE
(SIMP-3030) Add defines to generate JS rules

### PR DESCRIPTION
**Pull request by jeefberkey**
This commit adds a define for creating javascript polkit rules to be put
in `/etc/polkit-1/rules.d/`. It also adds a define for adding a
templated rule, with one basic condition and one basic response.

SIMP-3030 #close

### CODE REVIEW

**Comment by jeefberkey** [comment]
There is no approved polkit module on the Forge, so after this gets pushed I'd like to submit this to get approved.

File: manifests/authorization/basic_policy.pp [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+# // This file is managed by Puppet
+# polkit.addRule(function(action, subject) {
+#   if (<condition>) {
+#       return polkit.Result.<result>;
+#     }
+#   }
+# });
+# ```
+#
+# A user-specified <condition> can be supplied with the $condition parameter,
```
Comment (nick-markowski) [inline manifests/authorization/basic_policy.pp]: Good example

File: manifests/authorization/basic_policy.pp [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+#   polkit.addRule(function(action, subject) {
+#     if ((action.id == 'org.libvirt.unix.manage') && subject.user == 'testuser' && subject.isInGroup('testgroup') && subject.local && subject.active) {
+#         return polkit.Result.YES;
+#       }
+#     }
+#   });
+#
+# @param ensure Create or destroy the rules file
+#
+# @param result The end result of the polkit, like 'yes' or 'auth_admin'
```
Comment (nick-markowski) [inline manifests/authorization/basic_policy.pp]: :%s/like/for example

File: manifests/authorization/basic_policy.pp [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+#
+# @param action_id The polkit action to operate on
+#
+#   * A list of available actions can be found by running `pkaction`
+#
+# @param group Group to check membership of
```
Comment (nick-markowski) [inline manifests/authorization/basic_policy.pp]: The group(s) that a user belongs to

File: manifests/authorization/basic_policy.pp [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+#
+# @param user User to check
```
Comment (nick-markowski) [inline manifests/authorization/basic_policy.pp]: Specifies a user specific policy

You may want to put this before group

File: manifests/authorization/basic_policy.pp [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+#
+# @param local Check of the user is a local user. See man page for more
```
Comment (nick-markowski) [inline manifests/authorization/basic_policy.pp]: Check if

File: manifests/authorization/basic_policy.pp [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+  if !$condition {
+    if !$action_id {
+      fail('If $condition is not specified, $action_id must be')
+    }
+  }
+  if $facts['os']['release']['major'] == '6' {
+    fail('The version of Polkit available on EL6 does not support javascript configuration')
+  }
+
+  $_opts = {
```
Comment (nick-markowski) [inline manifests/authorization/basic_policy.pp]: Curious why you made condition a function. If you collapse `content` and `condition` into a (non-inline) template, you should be able to get rid of everything below, except rule would become

polkit::authorization::rule { $name:
     ensure   => $ensure,
     priority => $priority,
     rulesd   => $rulesd,
     content  => template("${module_name}/<some_template_with_content>.erb"),
   }

File: spec/defines/authorization/basic_policy_spec.rb [hunk 7]
```diff
@@ -0,0 +1,132 @@
+require 'spec_helper'
+
+describe 'polkit::authorization::basic_policy' do
+  supported_os = on_supported_os.delete_if { |e| e =~ /-6-/ } #TODO do this right
+  supported_os.each do |os, facts|
+  # on_supported_os.each do |os, facts|
```
Comment (nick-markowski) [inline spec/defines/authorization/basic_policy_spec.rb]: cruft

File: spec/defines/authorization/basic_policy_spec.rb [hunk 8]
```diff
@@ -0,0 +1,132 @@
...
+    context "on #{os}" do
+      let(:facts) do
+        facts
+      end
+
+      context 'authorize group to do an action' do
```
Comment (nick-markowski) [inline spec/defines/authorization/basic_policy_spec.rb]: Good tests

File: templates/basic_policy.epp [hunk 9]
```diff
@@ -0,0 +1,19 @@
+<%- |
+  String  $_condition,
```
Comment (op-ct) [inline templates/basic_policy.epp]: There's no need for the underscore prefix―it's a parameter.

File: spec/defines/authorization/rule_spec.rb [hunk 10]
```diff
@@ -0,0 +1,41 @@
...
+    context "on #{os}" do
+      let(:facts) do
+        facts
+      end
+
+      context 'with minimal parameters' do
+        let(:title) {'test'}
+        let(:params) {{
+          :ensure  => 'present',
+          :content => 'totally javascript trust me it me your friend'
```
Comment (op-ct) [inline spec/defines/authorization/rule_spec.rb]: Grammatical trustworthiness ~~confirmed~~ considered

File: manifests/authorization/basic_policy.pp (hunk 6 above)
Comment (op-ct) [inline manifests/authorization/basic_policy.pp]: I agree.  The function transforms data into content specific only to that template; the template should be responsible for rendering it.  (The functon is also simple enough that it doesn't need to be separate).

**Pull request merged**